asQ/allatonce/function.py

```diff
@@ -72,6 +72,16 @@ def __init__(self, ensemble, time_partition, function_space):
         self.function = fd.Function(self.function_space)
         self.initial_condition = fd.Function(self.field_function_space)
 
+        # Functions to view each timestep
```

Comment:
@connorjward please could you have a look over how I'm constructing `Function`s that view multiple components of a mixed `Function` here? I've essentially adapted the code from `firedrake.Function.subfunctions` to be able to make mixed subfunctions (so creating `Functions` using `pyop2.MixedDat` rather then `Dat`). A simplified example of what I'm doing:

```
# make a mixed function space for a single timestep:
V0 = fd.FunctionSpace(mesh, "CG", 1)
V1 = fd.FunctionSpace(mesh, "DG", 0)
V = V0*V1

# make a mixed function space to represent a timeseries of two timesteps:
# W = V0*V1*V0*V1
W = V*V 

# a Function for a timeseries of two timesteps
w = fd.Function(W)

# get the dats for the first timestep (the first two components of W)
dats = (w.subfunctions[i].dat for i in (0, 1))
# make a function to view the first timestep in w
wsub0 = fd.Function(V, val=MixedDat(dats))

# do the same for the second timestep
dats = (w.subfunctions[i].dat for i in (2, 3))
wsub1 = fd.Function(V, val=MixedDat(dats))
```
Is this the Right Way to do this? It passes the tests and seems to be behaving as I'd expect but is there likely to be any unexpected behaviour/pitfalls? Thanks!